apps/token/pint/token/src/contract.pnt

```diff
@@ -127,21 +154,39 @@ predicate Transfer {
     constraint @safe_increment(nonce);
     
     // Check the authorization predicate.
-    var auth_addr: PredicateAddress;
-    @auth(key; auth_addr; auth::@transfer(); auth::@transfer_with());
+    constraint match auth.mode {
+        TransferAuthMode::Signed(auth) => match auth.mode {
+            TransferSignedMode::All => @verify_key({key, to, amount, nonce'}; auth.sig; key),
+            TransferSignedMode::Key => @verify_key({key, nonce'}; auth.sig; key),
+            TransferSignedMode::KeyTo => @verify_key({key, to, nonce'}; auth.sig; key),
+            TransferSignedMode::KeyAmount => @verify_key({key, amount, nonce'}; auth.sig; key),
+        },
+        TransferAuthMode::Predicate(addr) => @check_if_predicate_owns(addr; key),
+    };
```

Comment:
This part may need verbal explanation, maybe we should mention somewhere why these exist, i.e. maybe the user wants to pay the solver but they don't know who the solver is so they don't sign over the `to`